### DISCUSSION TITLE
Portal Like Music for in spaceship

### DISCUSSION THREAD

**Discussion by TheShadowOfHassen** [endless-sky/endless-sky#8203] [Ideas]
This game is amazing, but when flying for long period times in my spaceship I'd really like some music. People have already talked a lot of it and I've seen some people even started working on some. I have an idea that might be a bit different.

It wouldn't work in the planets. For those you could have different songs based off of themes for the species and different effects depending on whether its a space station a city, farming, paradise or mining world (IE. Space station echoey synth, paradise worlds brass)

But in the spaceship I was thinking music like in the Game Portal, not in style but in Portal and Portal 2 they had adaptive music. (Although I like the portal sound and in some cases it might work) But with adaptive music when you shot a portal the music pitch changed, cubes give a different song and a special bit plays if you kill a turret. You get the point.

It'd take a bit more work but would cut down the annoyance of 15 min straight of a music loop. (a theme for the part of space) Different engines could alter the pitch of of the music so could flying near planets. Fighting would add a variant of the track with different sounds for shooting missiles or destroying a ship you get the point.

I don't know C++ so I can't help with the back end but if someone wants to experiment with this idea I can mess with the music. I'm not the best composer but I've done some.

**Comment by RisingLeaf** [comment]
I will link you to this: 
https://github.com/endless-sky/endless-sky/pull/7538

As you see music is already in work and we even have a creator who already created a bunch of tracks, we just need o implement them

**Comment by TheShadowOfHassen** [comment]
Ok got it, sorry. It was just an Idea I had. If the request gets merged I'll make certain to test it out. I can do music but I am not the best so It might be better for other people to do it.
